hack/test-stirunimage.sh

```diff
@@ -136,6 +136,11 @@ docker run test >& "${WORK_DIR}/s2i-override-run.log"
 grep "Running custom run" "${WORK_DIR}/s2i-override-run.log"
 check_result $? "${WORK_DIR}/s2i-override-run.log"
 
+test_debug "s2i build with add-host option"
+s2i build https://github.com/openshift/ruby-hello-world centos/ruby-23-centos7 --add-host rubygems.org:1.1.1.1 test-ruby-app &> "${WORK_DIR}/s2i-add-host.log"
+grep "Build failed" "${WORK_DIR}/s2i-add-host.log"
```

Comment:
Can this check for a more specific failure reason related to the rubygems.org host not being found?

Reply:
yes, but the error will change if 1.1.1.1 is down or close its http/https ports. I change the test scenario testing both cases----with and without --add-host flag---to make sure the failure is caused by --add-host flag